semanticdb/metacp/src/main/scala/scala/meta/internal/scalacp/SymbolOps.scala

```diff
@@ -1,16 +1,78 @@
 package scala.meta.internal.scalacp
 
 import java.util.{HashMap, HashSet}
+import scala.meta.internal.metacp._
+import scala.collection.mutable
 import scala.meta.internal.{semanticdb => s}
 import scala.meta.internal.semanticdb.Scala._
 import scala.meta.internal.semanticdb.Scala.{Descriptor => d}
-import scala.meta.internal.semanticdb.Scala.{Names => n}
 import scala.meta.internal.semanticdb.SymbolInformation.{Kind => k}
 import scala.tools.scalap.scalax.rules.scalasig._
 
 trait SymbolOps { self: Scalacp =>
   lazy val symbolCache = new HashMap[Symbol, String]
+  lazy val isJavaDefined = mutable.Map.empty[Symbol, Boolean]
   implicit class XtensionSymbolSSymbol(sym: Symbol) {
+
+    def isReallyPackage: Boolean = {
```

Comment:
Let's move these methods to `XtensionSymbol`. This implicit class, as well as its neighbor right below, are focused on high-level concepts from the SemanticDB spec.

Reply:
Done.